test/cypress/integration/worldpay-3ds-flex-ddc/payment-with-session-id.cy.test.js

```diff
@@ -189,27 +172,115 @@ describe('Worldpay 3ds flex card payment flow', () => {
     })
   })
 
-  setUpAndCheckCardPaymentPage()
   describe('DDC times out', () => {
-    it('iframe post should still submit the form but without the worldpaySessionId', () => {
+    it('Should submit the form from the iframe post without the worldpaySessionId ', () => {
+      cy.log('Should setup a standard payment page')
+
+      cy.task('setupStubs', createPaymentChargeStubsEnglish)
+      cy.visit(`/secure/${tokenId}`)
+
+      // 1. Charge will be created using this id as a token (GET)
+      // 2. Token will be deleted (DELETE)
+      // 3. Charge will be fetched (GET)
+      // 4. Service related to charge will be fetched (GET)
+      // 5. Charge status will be updated (PUT)
+      // 6. Client will be redirected to /card_details/:chargeId (304)
+      cy.location('pathname').should('eq', `/card_details/${chargeId}`)
+
+      cy.task('clearStubs')
+      cy.task('setupStubs', checkCardDetailsStubs)
+
+      cy.server()
+      cy.route('POST', `/check_card/${chargeId}`).as('checkCard')
+
+      cy.log('Should enter and validate a correct card')
+
+      cy.get('#card-no').type(validPayment.cardNumber)
+      cy.get('#card-no').blur()
+
+      // 7. Charge status will be fetched (GET)
+      // 8. CardID POST to verify that entered card is correct - this is configured to return valid
+      cy.wait('@checkCard')
+      cy.get('#card-no').should('not.have.class', 'govuk-input--error')
+
+      cy.log('Should enter payment details')
+
+      cy.get('#expiry-month').type(validPayment.expiryMonth)
+      cy.get('#expiry-year').type(validPayment.expiryYear)
+      cy.get('#cardholder-name').type(validPayment.name)
+      cy.get('#cvc').type(validPayment.securityCode)
+      cy.get('#address-line-1').type(validPayment.addressLine1)
+      cy.get('#address-city').type(validPayment.city)
+      cy.get('#address-postcode').type(validPayment.postcode)
+      cy.get('#email').type(validPayment.email)
+
       const confirmPaymentDetailsStubsPop = _.cloneDeep(confirmPaymentDetailsStubs)
       confirmPaymentDetailsStubsPop.pop()
 
+      cy.task('clearStubs')
       cy.task('setupStubs', confirmPaymentDetailsStubs)
 
+      cy.log('Should submit the form from the iframe post but without the worldpaySessionId')
+
       cy.get('#card-details').submit().should($form => {
         const formVal = $form.first()[0].elements.worldpay3dsFlexDdcResult
         expect(formVal).to.eq(undefined)
       })
-    })
-  })
 
-  setUpAndCheckCardPaymentPageWithGooglePay()
-  describe('Web payment text is not displayed on DDC spinner screen', () => {
-    it('iframe post should still submit the form but without the worldpaySessionId', () => {
+      cy.log('Should setup a Google pay payment page')
```

Comment:
Looks like this is specific to Google Pay payment and the code above is the card payment. Worth having a separate test for Google Pay payment.

Reply:
Just separated out the Google pay part of the test - but kept it in same `describe` block.

Appears that they are both testing the card payments - but this bit of the test is testing a `card payment` when `Google pay` is present.

Please take a look.

Reply:
Seemed to be an issue with headless mode:
-  Had to put it into it's own describe block.
- Test would fail if it was not setup with the standard stub, before being setup with the Google pay stub:
https://github.com/alphagov/pay-frontend/pull/3541/files#r1259295847

Please take a look.